Kernel/Devices/Audio/IntelHDA/OutputPath.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023, Jelle Raaijmakers <jelle@gmta.nl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/NonnullOwnPtr.h>
+#include <AK/NonnullRefPtr.h>
+#include <Kernel/Devices/Audio/Channel.h>
+#include <Kernel/Devices/Audio/IntelHDA/Stream.h>
+
+namespace Kernel::Audio::IntelHDA {
+
+class WidgetNode;
+
+class OutputPath {
+public:
+    static ErrorOr<NonnullOwnPtr<OutputPath>> create(NonnullRefPtr<WidgetNode> output_widget,
+        NonnullRefPtr<WidgetNode> pin_widget, NonnullOwnPtr<OutputStream> output_stream)
+    {
+        return adopt_nonnull_own_or_enomem(new (nothrow) OutputPath(output_widget, pin_widget, move(output_stream)));
+    }
+
+    NonnullRefPtr<WidgetNode> output_widget() { return m_output_widget; }
+    NonnullRefPtr<WidgetNode> pin_widget() { return m_pin_widget; }
+    OutputStream& output_stream() { return *m_output_stream; }
+
+    ErrorOr<void> set_format(u32 sample_rate, u8 pcm_bits, u8 number_of_channels)
+    {
+        // 7.3.3.8: Converter Format
+        // "The Converter Format control determines the format the converter will use. This must match the
+        // format programmed into the Stream Descriptor on the controller so that the data format being
+        // transmitted on the link matches what is expected by the consumer of the data."
+        TRY(m_output_stream->set_format(sample_rate, pcm_bits, number_of_channels));
+        TRY(m_output_widget->set_converter_format(sample_rate, pcm_bits, number_of_channels));
```

Comment:
Note to self: check with output widget which sample rates and bit sizes are actually supported.